PhysicsTools/NanoAOD/python/jets_cff.py

```diff
@@ -271,7 +272,8 @@
 for modifier in run2_miniAOD_80XLegacy, run2_nanoAOD_94X2016:
   modifier.toModify( jetTable.variables, jetId = Var("userInt('tightIdLepVeto')*4+userInt('tightId')*2+userInt('looseId')",int,doc="Jet ID flags bit1 is loose, bit2 is tight, bit3 is tightLepVeto"))
 run2_jme_2016.toModify( jetTable.variables, puIdDisc = Var("userFloat('pileupJetId:fullDiscriminant')",float,doc="Pilup ID discriminant with 80X (2016) training",precision=10))
-run2_jme_2017.toModify( jetTable.variables, puIdDisc = Var("userFloat('puId94XDisc')", float,doc="Pilup ID discriminant with 94X (2017) training",precision=10))
+run2_jme_2017.toModify( jetTable.variables, puId = Var("userInt('puId106XUL17Id')", int,doc="Pileup ID flags with 106X (2017) training"))
+run2_jme_2017.toModify( jetTable.variables, puIdDisc = Var("userFloat('puId106XUL17Disc')", float,doc="Pileup ID discriminant with 106X (2017) training",precision=10))
```

Comment:
@alefisico @lathomas 

while it's correct to update the UL 2017, we should keep the EOY untouched; after line 276 add this

for modifier in run2_nanoAOD_94XMiniAODv1, run2_nanoAOD_94XMiniAODv2:
   modifier.toModify( jetTable.variables, puIdDisc = Var("userFloat('puId94XDisc')", float,doc="Pilup ID discriminant with 94X (2017) training",precision=10))

What you want to do for the 2016,2018 of UL ? now you have the old id of the EOY` 
pileupJetId:fullDiscriminant` and `puId106XUL17Disc `

Reply:
Hello. Yes this is intended (we only have UL training for 2017 so far)

Reply:
ok for the UL-2016 and 2018 we can keep the old EOY or now.
But there is no reason to change the puId for the 2017 EOY to the new one since people will have in hands the SF for the EOY.

This block of code is still required after line 276
```
for modifier in run2_nanoAOD_94XMiniAODv1, run2_nanoAOD_94XMiniAODv2:
modifier.toModify( jetTable.variables, puIdDisc = Var("userFloat('puId94XDisc')", float,doc="Pilup ID discriminant with 94X (2017) training",precision=10))
```

Reply:
to summarize this is what we want:

EOY 16 (run2_nanoAOD_94X2016) pileupJetId:fullDiscriminant 
EOY 17 (run2_nanoAOD_94XMiniAODv1 + run2_nanoAOD_94XMiniAODv2 ) puId94XDisc 
EOY 18 (run2_nanoAOD_102Xv1) pileupJetId102X 
UL 16 (run2_nanoAOD_106Xv1, run2_jme_2016) pileupJetId:fullDiscriminant [temporary]
UL 17 (run2_nanoAOD_106Xv1, run2_jme_2017) puId106XUL17Disc 
UL 18 (run2_nanoAOD_106Xv1)  pileupJetId102X [temporary]

now how you do in reMiniAOD ?

Reply:
ok I just implemented this. I think for reMiniAOD we should use the same strategy as in the nanoAODv8 for UL.